scripts/tests/test_justfile_discoverability.py

```
@@ -0,0 +1,171 @@
+
+def _just_recipes() -> dict[str, dict[str, Any]]:
+    """Return recipe metadata keyed by recipe name."""
+    return cast("dict[str, dict[str, Any]]", _just_document()["recipes"])
+
+
+def _dependency_names(recipe: dict[str, Any]) -> set[str]:
+    """Return the direct recipe dependencies recorded by Just."""
+    return {dependency["recipe"] for dependency in recipe["dependencies"]}
+
+
+def _body_fragments(value: Any) -> list[str]:
```

Ruff (0.16.0)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)